pkg/hubble/server/serveroption/option.go

```diff
@@ -64,22 +64,28 @@ func WithTCPListener(address string) Option {
 // owner is set to socketGroup.
 func WithUnixSocketListener(path string) Option {
 	return func(o *Options) error {
+		if o.UnixSocketListener != nil {
```

Comment:
do we need to call o.UnixSocketListener.Close() in this if condition ?

Reply:
Looking only at this function, we either return an error leaving `o` in the same state we found it or we modify `o` with success. This is a clean behaviour and closing an already configured socket would break this promise.

As it is currently used, when this codepath is hit `pkg/hubble/server/server.NewServer` (i.e. the caller) would return a "failed to apply option" error and we'll loose ref to the `Options` struct (and thus the socket).

I would say we leave it as-is because it's corner case enough to let the garbage collector close the socket at some point. Alternatively, I would be in favour of a `func (o *Options) Cleanup()` to be called by `pkg/hubble/server/server.NewServer` on failure.

Reply:
thanks for your explanation, it's making sense to me :tada: